pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactedTopicImpl.java

```diff
@@ -311,9 +310,8 @@ private static int comparePositionAndMessageId(PositionImpl p, MessageIdData m)
             .compare(p.getEntryId(), m.getEntryId()).result();
     }
 
-    @VisibleForTesting
-    PositionImpl getCompactionHorizon() {
-        return this.compactionHorizon;
+    public synchronized Optional<Position> getCompactionHorizon() {
```

Comment:
Why should we use `synchronized ` ?

Reply:
Another thread update the `compactionHorizon` https://github.com/apache/pulsar/blob/07ef9231db8b844586b9217ee2d59237eb9c54b7/pulsar-broker/src/main/java/org/apache/pulsar/compaction/CompactedTopicImpl.java#L70